src/passport-saml/strategy.ts

```diff
@@ -1,131 +1,131 @@
-import * as passport from 'passport-strategy';
-import * as util from 'util';
+import { Strategy as PassportStrategy } from 'passport-strategy';
 import * as saml from './saml';
 import * as url from 'url';
-import { AuthenticateOptions, AuthorizeOptions, SamlConfig, VerifyWithoutRequest, VerifyWithRequest } from './types';
-import type { Request } from 'express';
+import { AuthenticateOptions, AuthorizeOptions, RequestWithUser, SamlConfig, VerifyWithoutRequest, VerifyWithRequest } from './types';
 import { Profile } from './types';
 
-interface SAMLRequest extends Request {
-  samlLogoutRequest?: Profile
-}
-
-function Strategy (options: SamlConfig, verify: VerifyWithRequest | VerifyWithoutRequest) {
-  if (typeof options == 'function') {
-    verify = options;
-    options = {};
-  }
-
-  if (!verify) {
-    throw new Error('SAML authentication strategy requires a verify function');
-  }
-
-  // Customizing the name can be useful to support multiple SAML configurations at the same time.
-  // Unlike other options, this one gets deleted instead of passed along.
-  if  (options.name) {
-    this.name  = options.name;
-  }
-  else {
-    this.name = 'saml';
-  }
+class Strategy extends PassportStrategy {
+  name: string;
+  _verify: VerifyWithRequest | VerifyWithoutRequest;
+  _saml: saml.SAML;
+  _passReqToCallback?: boolean;
+  _authnRequestBinding?: string;
+
+  constructor(options: SamlConfig, verify: VerifyWithRequest | VerifyWithoutRequest) {
+    super();
+    if (typeof options == 'function') {
+      verify = options;
+      options = {};
+    }
 
-  passport.Strategy.call(this);
+    if (!verify) {
+      throw new Error('SAML authentication strategy requires a verify function');
+    }
 
-  this._verify = verify;
-  this._saml = new saml.SAML(options);
-  this._passReqToCallback = !!options.passReqToCallback;
-  this._authnRequestBinding = options.authnRequestBinding || 'HTTP-Redirect';
-}
+    // Customizing the name can be useful to support multiple SAML configurations at the same time.
+    // Unlike other options, this one gets deleted instead of passed along.
+    if  (options.name) {
+      this.name  = options.name;
+    }
+    else {
```

Comment:
We use different style `else` in this project, even in this single file. Can we adjust our linter to catch/fix this? I personally like `} else {`, but as long as it is consistent.

Reply:
My preference would to enforce code formatting with `eslint` or `prettier`. 

I like the idea of using `prettier` for code formatting and `eslint for code logic conventions`.

Automating code format checking can be especially valuable for open source projects that receive a lot of pull requests from new contributors who are not familiar with the conventions of the project.  Improving code formatting automation can be a separate issue and PR, though!